### PR TITLE
added dark mode switch

### PR DESCRIPTION
**Pull request by rabin245**
used provider as state manager and shared preference to save the theme for next time app opens

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: first
-description: A new Flutter project.
+description: Flutter application to calculate mod
```
Comment (anwesh-b) [inline pubspec.yaml:2]: Update version as well

File: lib/app/app.dart [hunk 2]
```diff
@@ -1,24 +1,55 @@
...
+      body: ModCalculator(),
+    );
+  }
+
+  StatefulBuilder buildSwitch() {
+    return StatefulBuilder(
+      builder: (BuildContext context, StateSetter setState) {
+        return Switch(
+          value: context.read<ThemeProvider>().isDarkMode,
+          // value: Provider.of<ThemeProvider>(context).isDarkMode,
```
Comment (anwesh-b) [inline lib/app/app.dart:44]: Remove if not needed

File: lib/app/app.dart [hunk 3]
```diff
@@ -1,24 +1,55 @@
...
+          onChanged: (value) {
+            context.read<ThemeProvider>().toggle(value);
+            // Provider.of<ThemeProvider>(context, listen: false)
+            //     .toggle(value);
```
Comment (anwesh-b) [inline lib/app/app.dart:48]: Remove if not needed

**Review by anwesh-b — CHANGES_REQUESTED** [review]

File: lib/app/app.dart (hunk 2 above)
Comment (rabin245) [reply lib/app/app.dart:44]: that is for learning purpose. Provider package still has backward compatibility and the comment version also works which is much more easier to understand and learn. Tei timi harulai bujhna huncha vanera rakheko tyo provider ko chai

Comment (anwesh-b) [reply lib/app/app.dart:44]: Okay :)

**Review by anwesh-b — APPROVED** [review]

**Pull request merged**